### PR TITLE
Feature -  5469 - Request confirmation page

### LINKED ISSUE #5469: ✨ (Search Page) Confirmation page
## ✨ Feature

Search page needs a confirmation page to assure managers that their request was submitted successfully as well as giving them some information useful for troubleshooting.

## 🕵️ Details

Create a confirmation page that follows the successful submission of a request form. 
- Request ID/KEY might be the only thing worth looking into, I'm unsure if we have something we can use for this but is important to give them a unique identifier they can use for troubleshooting.
- the print button will just print this page as pdf. 
- submit a new request takes the user back to an empty search page.

## 🎨 Design File

[Link to figma](https://www.figma.com/proto/UsrB15yicgDCLDQmx3snyl/User-Flows?page-id=3513%3A11410&node-id=3513%3A15087&viewport=133%2C584%2C0.92&scaling=min-zoom&starting-point-node-id=3537%3A11514)

## 📸 Screenshot

image

## 🙋‍♀️ Proposed Implementation

(optional) Do you have a proposed implementation?

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

- [ ] This is a new page, available at `/search/request/<request-id>`
- [ ] It shows mostly static text, but also the id of the request you just submitted
- [ ] It has a print button that opens the default browser print dialog (no special view, just default print)
- [ ] It has a button that redirects to the empty Search page
- [ ] After requesting a talent search, you are directed to this page
- [ ] If someone navigates here by different method, just show the page normally, don't verify or guard
- [ ] Breadcrumb destinations: homepage \ search page \ this page

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5469

## 👋 Introduction

This adds the request confirmation page and redirects to it upon successful request submission.

## 🕵️ Details

The design looks slightly different than mockup for a few minor a11y reasons:

- Removed the bullets since the text is not listing anything
- The hero is using our standardized hero where the one in the mockup is slightly different

## 📋 To Do

- [x] Send for translation
- [x] Add translations

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build the app `npm run build`
2. Navigate to the search page `/search`
3. Apply some filters and start a request
4. Submit the request
5. Confirm your are redirected to the new page on _successful_ submission
6. Confirm the page matches the mockup closely (keeping in mind the details mentioned previously)
7. Confirm the ID matches the request ID
8. Confirm the print button works as expected
9. Confirm all links on the page go to the expected pages

## 📸 Screenshot

localhost_8000_en_search

### CODE REVIEW

File: apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            id: "ltn4Er",
+            description:
+              "Description of when the user should expect a response to their request",
+          })}
+        </Text>
+        <Text>
+          {intl.formatMessage(
+            {
+              defaultMessage:
+                "If you have not heard from us or have any immediate questions and request further guidance, please get in touch with use at: <mailLink>recruitmentimit-recrutementgiti@tbs-sct.gc.ca</mailLink>",
```
Comment (vd1992) [inline apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx:124]: please get in touch with us*
might need to double check the corresponding translation too

**Review by vd1992 — CHANGES_REQUESTED** [review]

File: apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx]: Good eye! Fixed in [fix type on en request confirmation page](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5539/commits/d05bdf0a2c92127df6ff3600a9fc9995885c1e2a). FR looked good but maybe @mnigh could confirm?

Comment (esizer) [reply apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx]: heh, typo in the commit message to fix a typo 😓

Comment (vd1992) [reply apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx]: passes the google check at least

image

**Review by vd1992 — APPROVED** [review]
things look good on my end, great stuff as usual!

only thing of note to me is that you can go to whatever request you want like say
`/request/+&asdf`
and it'll generate a page, but since this is currently a public page with no authentication or database interaction, it doesn't merit worry yet

**Comment by esizer** [comment]
> only thing of note to me is that you can go to whatever request you want like say
> `/request/+&asdf`
> and it'll generate a page, but since this is currently a public page with no authentication or database interaction, it doesn't merit worry yet

Heh, yeah. That is kind of why I brought up the question about how we determine if the request itself exists etc. 

Does this warrant adding a small API wrapper around this to get the request and throw a 404 if we don't have one with the matching ID @tristan-orourke ? Or do we even care if people can get there with any param added on?

**Comment by vd1992** [comment]
> > only thing of note to me is that you can go to whatever request you want like say
> > `/request/+&asdf`
> > and it'll generate a page, but since this is currently a public page with no authentication or database interaction, it doesn't merit worry yet
> 
> Heh, yeah. That is kind of why I brought up the question about how we determine if the request itself exists etc.
> 
> Does this warrant adding a small API wrapper around this to get the request and throw a 404 if we don't have one with the matching ID @tristan-orourke ? Or do we even care if people can get there with any param added on?

currently there isn't any information to protect on the page is there?
the request ID is simply the URL chunk

**Comment by tristan-orourke** [comment]
Haha, I find this kind of hilarious, we don't actually need to load any information about the request so the id in the url is the only data the page needs 😆 . 

Hmmm... I don't feel strongly either way. I think it's fine as it is, for now. We can add the api wrapper when and if we actually need any data about the request.

**Comment by esizer** [comment]
> Haha, I find this kind of hilarious, we don't actually need to load any information about the request so the id in the url is the only data the page needs 😆 .
> 
> Hmmm... I don't feel strongly either way. I think it's fine as it is, for now. We can add the api wrapper when and if we actually need any data about the request.

Sounds good to me! I just used the param directly for speed and reducing the number of requests to API. Seems kinda silly to send a request just to display the exact same info 😜 And, even if people figure out you can render that directly to the page, I think the worst thing that could happen would be people putting naughty words in the URL to have them appear on page since react is escaped by default 😉

**Comment by tristan-orourke** [comment]
>  And, even if people figure out you can render that directly to the page, I think the worst thing that could happen would be people putting naughty words in the URL to have them appear on page since react is escaped by default 😉

There's our easter egg!
Let's render something special if you enter the konami code! 😂

File: apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/pages/SearchRequests/RequestConfirmationPage/RequestConfirmationPage.tsx]: I think Google Translate autocorrects and hides the errors. It does not pass my test! 😆 

At the very least, the part _de nos Nouvelles_ seems wrong, especially the capital N. The sentence in English is also awkward, I would recommend having it re-written and then translated again in French.

**Pull request merged**